src/main/navigation/navigation_fw_launch.c

```diff
@@ -316,7 +318,7 @@ static fixedWingLaunchEvent_t fwLaunchState_FW_LAUNCH_STATE_WAIT_DETECTION(timeU
 
     const bool isBungeeLaunched = isForwardAccelerationHigh && isAircraftAlmostLevel;
     const bool isSwingLaunched = (swingVelocity > navConfig()->fw.launch_velocity_thresh) && (imuMeasuredAccelBF.x > 0);
-    const bool isForwardLaunched = (gpsSol.groundSpeed > navConfig()->fw.launch_velocity_thresh) && (imuMeasuredAccelBF.x > 0);
+    const bool isForwardLaunched = isGPSHeadingValid() && (imuMeasuredAccelBF.x > 0);
```

Comment:
I would still do an **additional** explicit velocity check to make sure that if at some point `isGPSHeadingValid` logic is changed, we won't run into issues with repurposing it for launch detection.

I hope this will be the last one :laughing:

Reply:
Sure thing 😉